server/types/mock.go

```diff
@@ -163,10 +164,15 @@ func (mp MockProxy) Redirect(req Request) (*MockResponse, error) {
 	if err != nil {
 		return nil, err
 	}
-	proxyReq.Header = req.Headers
+	proxyReq.Header = req.Headers.Clone()
 	if mp.KeepHost {
 		proxyReq.Host = req.Headers.Get("Host")
 	}
+	for key, values := range mp.Headers {
+		for _, value := range values {
+			proxyReq.Header.Add(key, value)
```

Comment:
By doing that if a header is already setted in request, the value of proxy header will be added and will not replace it.
And I don't think it's the desired behavior

Reply:
```go
proxyReq.Header[key] = values
```

Reply:
I added `proxyReq.Header.del(key)` in the keys loop to clear any present header. `proxyReq.Header[key] = values` bypasses header key canonicalization. As a consequence, header may not be overwritten if key is specified with a different case in proxy definition and in incoming request.